### DISCUSSION TITLE
如何更优雅的处理表单联动，异步获取数据，并且回填的问题

### DISCUSSION THREAD

**Discussion by semi-xi** [ant-design/ant-design#33886] [General]
在日常开发中后台的时候，表单的联动是必不可少，很多时候都是体现在表单A的值->引起表单B的值显示或者隐藏，这里我们尚可用`shouldUpdate` 来设置是否隐藏。
但是如果升级一下呢，表单A->表单B的显示隐藏，表单B的初始化值是通过异步获取的，这个时候应该怎么处理呢？直接在`render B`的时候，await这个请求，然后再return 这个组件出去？
这个也只是普通模式，如果是编辑模式呢？
就是说存在表单A->表单B的值，这个时候直接stFields是设置不了B的，因为B需要在A存在的时候才存在，这种又是怎么解决呢。我找了很久都没找到一个很好的解决方式

**Comment by semi-xi** [comment]
呃~~好像没人

**Reply by huntdream** [reply]
应该可以在B组件挂载的时候调接口获取数据然后set

**Reply by semi-xi** [reply]
以前我们也试过这种，但是当处于编辑场景的时候，这里就变得更加不可控了。或者说当C组件依赖B组件的值的时候，就会陷入了很麻烦的组件通信管理

**Reply by huntdream** [reply]
能否放在同一个Form里面？

**Comment by yoyo837** [comment]
加个变量控制就行了

**Comment by yoyo837** [comment]
控制组件渲染，

**Reply by semi-xi** [reply]
但是你想过没有，如果当你处于编辑状态的时候，你这个时候就需要setState这个变量，让他显示出来，再setFields去设置这个值了。实际上这里没有特别好的控制流程方法。当这种场景一多的时候就会变得很难维护了

**Comment by yoyo837** [comment]
我个人也有大量这种场景，加载数据就loading咯，loading好了就该干嘛就干嘛。
